Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -5,128 +5,34 @@
-        protected static async void CreateCopyProfileValues()
-        {
-            profileToCopy = profile;
-            ScriptableObject newProfile = CreateInstance(profile.GetType().ToString());
-            profile = newProfile.CreateAsset("Assets/MixedRealityToolkit-Generated/CustomProfiles") as BaseMixedRealityProfile;
-            Debug.Assert(profile != null);
-
-            await new WaitUntil(() => profileToCopy != profile);
-
-            Selection.activeObject = null;
-            PasteProfileValues();
-            Selection.activeObject = profile;
-
-            if (!profileToCopy.IsCustomProfile)
-            {
-                // For now we only replace it if it's the master configuration profile.
-                // Sub-profiles are easy to update in the master configuration inspector.
-                if (MixedRealityToolkit.Instance.ActiveProfile.GetType() == profile.GetType())
-                {
-                    MixedRealityToolkit.Instance.ActiveProfile = profile as MixedRealityToolkitConfigurationProfile;
-                }
-            }
-        }
-
-        [MenuItem("CONTEXT/BaseMixedRealityProfile/Copy Profile Values", false, 1)]
-        private static void CopyProfileValues()
-        {
-            profileToCopy = profile;
-        }
-
-        [MenuItem("CONTEXT/BaseMixedRealityProfile/Paste Profile Values", true)]
-        private static bool PasteProfileValuesValidation()
-        {
-            return profile != null &&
-                   targetProfile != null &&
-                   profileToCopy != null &&
-                   targetProfile.FindProperty(IsCustomProfileProperty).boolValue &&
-                   profile.GetType() == profileToCopy.GetType();
-        }
-
-        [MenuItem("CONTEXT/BaseMixedRealityProfile/Paste Profile Values", false, 2)]
-        private static void PasteProfileValues()
-        {
-            Undo.RecordObject(profile, "Paste Profile Values");
-            bool targetIsCustom = targetProfile.FindProperty(IsCustomProfileProperty).boolValue;
-            string originalName = targetProfile.targetObject.name;
-            EditorUtility.CopySerialized(profileToCopy, targetProfile.targetObject);
-            targetProfile.Update();
-            targetProfile.FindProperty(IsCustomProfileProperty).boolValue = targetIsCustom;
-            targetProfile.ApplyModifiedProperties();
-            targetProfile.targetObject.name = originalName;
-            Debug.Assert(targetProfile.FindProperty(IsCustomProfileProperty).boolValue == targetIsCustom);
-            AssetDatabase.SaveAssets();
-        }
-
-        /// <summary>
-        /// Render the Mixed Reality Toolkit Logo.
-        /// </summary>
-        protected void RenderMixedRealityToolkitLogo()
+        [Obsolete("Use MixedRealityInspectorUtility.CheckMixedRealityConfigured instead")]
+        public static bool CheckMixedRealityConfigured(bool flag = true)
```

Comment:
It's not depreciated, it was moved into a utility class because not all inspector scripts will inherit from this one, but we needed an easy way to check if the configuration was valid.

Reply:
The refactoring makes sense, I am more curious on our public facing story for refactoring/moving code. We can either align with the idea that all existing functions should still exist in some form of the code base to provide information on how to migrate code. We can make previously existing functionality work for multiple releases in order to ease this process. Or we can remove things entirely that do not work anymore. Being new to the project, I'm curious what the consensus was. :)